lib/xcpretty/parser.rb

```diff
@@ -221,7 +221,7 @@ module Warnings
       # $1 = file_path
       # $2 = file_name
       # $3 = reason
-      COMPILE_WARNING_MATCHER = /^(\/.+\/(.*):.*:.*):\swarning:\s(.*)$/.freeze
+      COMPILE_WARNING_MATCHER = /^(\/.+\/([^:]*?):\d*?:\d*?):\swarning:\s(.*)$/.freeze
```

Comment:
1. Replace `.+` with `[^\/]`. It will take a lot less steps according to regex101 🚀 
2. You don't need all the `?` because it will anyway stop at the first character that's not in the regex.

Same for other places.

Reply:
1. (been there) produces the wrong result.
2. Done